src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -164,6 +164,10 @@ private HttpRequestMessage CreateHttpRequestMessage(Uri uri)
                 }
             }
 
+            // If the lastEventId was provided, include it as a header to the API request.
+            if (!string.IsNullOrWhiteSpace(lastEventId) && !request.Headers.Contains(Constants.LastEventIdHttpHeader))
+                request.Headers.Add(Constants.LastEventIdHttpHeader, lastEventId);
+
```

Comment:
I think this is _slightly_ wrong, in that it will _not_ override any `Last-Event-Id` header that was provided in the configuration.

Here's a scenario:

* The application creates an EventSource, and specifies in the configuration that it should start with a header of `Last-Event-Id: 123`. So, on the first connection attempt, it sends that header.
* It then receives several events from the stream, and they each have an `id` field. The last one received has `id: 131`. So the private field `_lastEventId` is now "131".
* Now there's a connection error, so we need to reconnect.

I think that in this scenario, it would be correct for it to send `Last-Event-Id: 131`, not `Last-Event-Id: 123`. The configuration specified what the application thought the last event was at _that_ time, but the state has changed since then. So I think on lines 168-169, it would be appropriate for it to overwrite any previously specified value for this header.